.github/workflows/codeql.yml

```diff
@@ -0,0 +1,41 @@
+name: "CodeQL"
+
+on:
+  push:
+    branches: [ "main" ]
+  pull_request:
+    # The branches below must be a subset of the branches above
+    branches: [ "main" ]
+    
+jobs:
+  analyze:
+    name: Analyze
+    runs-on: ubuntu-latest
+    permissions:
+      actions: read
+      contents: read
+      security-events: write
+
+    strategy:
+      fail-fast: false
+      matrix:
+        language: [ 'go' ]
+
+    steps:
+    - name: Checkout repository
+      uses: actions/checkout@v3
+
+    # Initializes the CodeQL tools for scanning.
+    - name: Initialize CodeQL
+      uses: github/codeql-action/init@v2
+      with:
+        languages: ${{ matrix.language }}
+
+
+    - name: Autobuild
```

Comment:
Does this work on this codebase, as all the Go code is under `bot`?

Reply:
I was unsure if it would either, but when the scan ran it looked like it was able to scan enough go to pass my sniff test: https://github.com/gravitational/shared-workflows/actions/runs/3516375568/jobs/5892855118#step:5:372
 
```
|                 Metric                 | Value |
+----------------------------------------+-------+
| Total lines of Go code in the database |  1670 |
```

Let me know if you this sounds off to you at all, thank you!